examples/gif_paintable/gif_paintable/frame.rs

```diff
@@ -0,0 +1,53 @@
+use std::time::Duration;
+
+use gtk::prelude::*;
+use gtk::{gdk, glib};
+
+pub struct Frame {
+    pub width: i32,
+    pub height: i32,
+    pub texture: gdk::Paintable,
+    pub frame_duration: Duration,
+}
+
+impl Frame {
+    pub fn new(f: image::Frame) -> Self {
+        let mut frame_duration = Duration::from(f.delay());
+
+        // convention is to use 100 milliseconds duration if it is defined as 0.
+        if frame_duration.is_zero() {
+            frame_duration = Duration::from_millis(100);
+        }
+
+        let image = f.into_buffer();
+
+        let samples = image.into_flat_samples();
```

Comment:
I think you need to ensure somewhere that this is actually `P: RGBA` and not anything else?

Reply:
What do you mean? `Frame::into_buffer()` returns an `RgbaImage` so we already know that `P: Rgba`.

Reply:
Ah I didn't realize it would return an `RgbaImage`. All good then, I assumed it's returning a generic thing.